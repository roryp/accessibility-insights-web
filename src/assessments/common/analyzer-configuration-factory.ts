// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Messages } from 'common/messages';
import { RuleAnalyzerConfiguration } from 'injected/analyzers/analyzer';
import { ScannerUtils } from 'injected/scanner-utils';

export class AnalyzerConfigurationFactory {
    public static forScanner(base: Partial<RuleAnalyzerConfiguration>): RuleAnalyzerConfiguration {
        const defaultValues: Partial<RuleAnalyzerConfiguration> = {
            resultProcessor: (scanner: ScannerUtils) => scanner.getAllCompletedInstances,
            telemetryProcessor: telemetryFactory => telemetryFactory.forAssessmentRequirementScan,
            analyzerMessageType: Messages.Assessment.AssessmentScanCompleted,
        };

        return {
            ...defaultValues,
            ...base,
        } as RuleAnalyzerConfiguration;
    }
}
