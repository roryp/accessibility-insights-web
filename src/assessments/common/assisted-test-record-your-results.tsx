// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import * as Markup from '../markup';

export const AssistedTestRecordYourResults = () => (
    <li>
        Record your results:
        <ol>
            <li>
                Select <Markup.Term>Fail</Markup.Term> for any instances that do not meet the
                requirement.
            </li>
            <li>
                Otherwise, select <Markup.Term>Pass</Markup.Term>. Or, after you have marked all
                failures, select <Markup.Term>Pass unmarked instances.</Markup.Term>
            </li>
        </ol>
    </li>
);
