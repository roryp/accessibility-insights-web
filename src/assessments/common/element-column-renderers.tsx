// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { AssessmentInstanceDetailsColumn } from 'DetailsView/components/assessment-instance-details-column';
import * as React from 'react';

export function onRenderPathColumn(item: InstanceTableRow): JSX.Element {
    let textContent = '';
    if (item.instance.target) {
        textContent = item.instance.target.join(';');
    }

    return (
        <AssessmentInstanceDetailsColumn
            background={null}
            labelText={null}
            textContent={textContent}
            tooltipId={null}
            customClassName="not-applicable"
        />
    );
}

export function onRenderSnippetColumn(item: InstanceTableRow): JSX.Element {
    return (
        <AssessmentInstanceDetailsColumn
            background={null}
            labelText={null}
            textContent={item.instance.html}
            tooltipId={null}
            customClassName="not-applicable"
        />
    );
}
