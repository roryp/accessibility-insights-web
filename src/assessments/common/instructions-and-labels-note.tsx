// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';

import * as Markup from '../markup';

export const InstructionsAndLabelsNotes = () => (
    <Markup.Emphasis>
        Note: For WCAG 2.1 compliance, both visible instructions and labels must be programmatically
        related to a widget. For WCAG 2.0 compliance, only visible instructions must be
        programmatically related to a widget.
    </Markup.Emphasis>
);
