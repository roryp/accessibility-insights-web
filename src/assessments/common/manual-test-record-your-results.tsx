// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import * as Markup from '../markup';

export interface ManualTestRecordYourResultsProps {
    isMultipleFailurePossible: boolean;
}

export const ManualTestRecordYourResults = (props: ManualTestRecordYourResultsProps) => (
    <li>
        Record your results:
        <ol>
            <li>
                If you find {props.isMultipleFailurePossible ? 'any failures' : 'a failure'}, select{' '}
                <Markup.Term>Fail</Markup.Term>,
                {props.isMultipleFailurePossible
                    ? ' then add them as failure instances'
                    : ' then add the failure instance'}
                .
            </li>
            <li>
                Otherwise, select <Markup.Term>Pass</Markup.Term>.
            </li>
        </ol>
    </li>
);
