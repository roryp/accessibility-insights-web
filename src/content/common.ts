// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as react from 'react';

import { ContentCreator } from 'views/content/content-page';
import { GuidanceTitle as gt } from 'views/content/guidance-title';
import { link } from './link';

export const create = ContentCreator(link);
export const React = react;
export const GuidanceTitle = gt;
