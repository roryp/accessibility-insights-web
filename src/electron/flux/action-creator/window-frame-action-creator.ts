// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { WindowFrameActions } from 'electron/flux/action/window-frame-actions';
import { SetSizePayload } from 'electron/flux/action/window-frame-actions-payloads';

export class WindowFrameActionCreator {
    constructor(private readonly windowFrameActions: WindowFrameActions) {}

    public setWindowSize(size: SetSizePayload): void {
        this.windowFrameActions.setWindowSize.invoke(size);
    }

    public maximize(): void {
        this.windowFrameActions.maximize.invoke();
    }

    public minimize(): void {
        this.windowFrameActions.minimize.invoke();
    }
    public restore(): void {
        this.windowFrameActions.restore.invoke();
    }

    public close(): void {
        this.windowFrameActions.close.invoke();
    }
}
