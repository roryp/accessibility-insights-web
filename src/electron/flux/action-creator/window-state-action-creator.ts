// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { RoutePayload } from '../action/route-payloads';
import { WindowStateActions } from '../action/window-state-actions';
import { WindowStatePayload } from '../action/window-state-payload';
import { WindowFrameActionCreator } from './window-frame-action-creator';

export class WindowStateActionCreator {
    constructor(
        private readonly windowStateActions: WindowStateActions,
        private readonly windowFrameActionCreator: WindowFrameActionCreator,
    ) {}

    public setRoute(payload: RoutePayload): void {
        this.windowStateActions.setRoute.invoke(payload);

        if (payload.routeId === 'deviceConnectView') {
            this.windowFrameActionCreator.setWindowSize({ width: 600, height: 391 });
        }
    }

    public setWindowState(payload: WindowStatePayload): void {
        this.windowStateActions.setWindowState.invoke(payload);
    }
}
