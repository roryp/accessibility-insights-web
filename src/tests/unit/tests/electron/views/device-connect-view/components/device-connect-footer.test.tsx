// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { WindowFrameActionCreator } from 'electron/flux/action-creator/window-frame-action-creator';
import { WindowStateActionCreator } from 'electron/flux/action-creator/window-state-action-creator';
import {
    DeviceConnectFooter,
    DeviceConnectFooterDeps,
    DeviceConnectFooterProps,
} from 'electron/views/device-connect-view/components/device-connect-footer';
import {
    footerButtonCancel,
    footerButtonStart,
} from 'electron/views/device-connect-view/components/device-connect-footer.scss';
import { shallow } from 'enzyme';
import { Button } from 'office-ui-fabric-react';
import * as React from 'react';
import { EventStubFactory } from 'tests/unit/common/event-stub-factory';
import { IMock, Mock, Times } from 'typemoq';

describe('DeviceConnectFooterTest', () => {
    let windowStateActionCreatorMock: IMock<WindowStateActionCreator>;
    let windowFrameActionCreatorMock: IMock<WindowFrameActionCreator>;
    let deps: DeviceConnectFooterDeps;

    const onClickMock = Mock.ofInstance(() => {});
    const eventStub = new EventStubFactory().createMouseClickEvent() as React.MouseEvent<Button>;

    beforeEach(() => {
        windowStateActionCreatorMock = Mock.ofType(WindowStateActionCreator);
        windowFrameActionCreatorMock = Mock.ofType(WindowFrameActionCreator);
        deps = {
            windowStateActionCreator: windowStateActionCreatorMock.object,
            windowFrameActionCreator: windowFrameActionCreatorMock.object,
        };
    });

    test('render', () => {
        const props: DeviceConnectFooterProps = {
            cancelClick: () => {
                return;
            },
            canStartTesting: false,
            deps,
        };
        const rendered = shallow(<DeviceConnectFooter {...props} />);

        expect(rendered.getElement()).toMatchSnapshot();
    });

    test('cancel click', () => {
        const props: DeviceConnectFooterProps = {
            cancelClick: onClickMock.object,
            canStartTesting: false,
            deps,
        };

        const rendered = shallow(<DeviceConnectFooter {...props} />);
        const button = rendered.find(`.${footerButtonCancel}`);
        button.simulate('click', eventStub);

        onClickMock.verify(onClick => onClick(), Times.once());
    });

    test('start testing changes route & maximizes window', () => {
        windowStateActionCreatorMock
            .setup(w => w.setRoute({ routeId: 'resultsView' }))
            .verifiable(Times.once());
        windowFrameActionCreatorMock.setup(w => w.maximize()).verifiable(Times.once());
        const props: DeviceConnectFooterProps = {
            cancelClick: onClickMock.object,
            canStartTesting: true,
            deps,
        };
        const rendered = shallow(<DeviceConnectFooter {...props} />);
        const button = rendered.find(`.${footerButtonStart}`);

        button.simulate('click', eventStub);

        windowStateActionCreatorMock.verifyAll();
        windowFrameActionCreatorMock.verifyAll();
    });
});
