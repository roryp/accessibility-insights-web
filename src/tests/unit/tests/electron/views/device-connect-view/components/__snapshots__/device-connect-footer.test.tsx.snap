// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceConnectFooterTest render 1`] = `
<footer
  className="deviceConnectFooter"
>
  <CustomizedDefaultButton
    className="footerButtonCancel"
    data-automation-id="device-connect-footer-cancel"
    onClick={[Function]}
    text="Cancel"
  />
  <CustomizedPrimaryButton
    className="footerButtonStart"
    data-automation-id="device-connect-footer-start"
    disabled={true}
    onClick={[Function]}
    text="Start testing"
  />
</footer>
`;
